account-kit/smart-contracts/src/ma-v2/account/modularAccountV2.ts

```diff
@@ -87,38 +109,69 @@ export async function createModularAccountV2<
  * });
  * ```
  *
- * @param {CreateModularAccountV2Params} config Configuration parameters for creating a Modular Account V2.
- * @returns {Promise<ModularAccountV2>} A promise that resolves to an `ModularAccountV2` providing methods for nonce retrieval, transaction execution, and more.
+ * @param {CreateModularAccountV2Params | CreateModularAccountV2ParamsNoSigner} config Configuration parameters for creating a Modular Account V2.
+ * @returns {Promise<ModularAccountV2 | ModularAccountV2NoSigner>} A promise that resolves to an `ModularAccountV2` providing methods for nonce retrieval, transaction execution, and more.
  */
-export async function createModularAccountV2(
-  config: CreateModularAccountV2Params
-): Promise<ModularAccountV2> {
+export async function createModularAccountV2<
+  TTransport extends Transport = Transport,
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+>(
+  config:
```

Comment:
overall I am unsure how to handle the parameters that are unique to each param set (no signer, webauthn account params and signer for `CreateModularAccountV2Params` and `CreateModularAccountV2ParamsNoSigner` respectively)